linera-core/src/unit_tests/test_helpers.rs

```diff
@@ -1,57 +1,90 @@
 // Copyright (c) Zefchain Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-#[cfg(test)]
-use assert_matches::assert_matches;
 use linera_chain::{ChainError, ChainExecutionContext};
 use linera_execution::{ExecutionError, SystemExecutionError};
 
 use crate::{client::ChainClientError, local_node::LocalNodeError, worker::WorkerError};
 
-#[cfg(test)]
+/// Asserts that an error is due to insufficient funding during an operation.
 pub fn assert_insufficient_funding_during_operation<T>(
     obtained_error: &Result<T, ChainClientError>,
     operation_index: u32,
 ) {
-    assert_matches!(
-        obtained_error.as_ref().err().unwrap(),
+    let error = match obtained_error.as_ref().err().unwrap() {
         ChainClientError::LocalNodeError(LocalNodeError::WorkerError(
-            WorkerError::ChainError(error)
-        )) if matches!(&**error, ChainError::ExecutionError(
-            execution_error, ChainExecutionContext::Operation(index)
-        ) if index == &operation_index && matches!(**execution_error, ExecutionError::SystemError(
-            SystemExecutionError::InsufficientFunding { .. }
-        )))
-    );
+            WorkerError::ChainError(error),
+        )) => error,
+        _ => panic!("Expected a ChainClientError::LocalNodeError with a WorkerError::ChainError"),
+    };
+
+    let (execution_error, context) = match &**error {
+        ChainError::ExecutionError(execution_error, context) => (execution_error, context),
+        _ => panic!("Expected a ChainError::ExecutionError"),
+    };
+
+    if let ChainExecutionContext::Operation(index) = context {
+        assert_eq!(index, &operation_index, "Operation index mismatch");
+    } else {
+        panic!("Expected ChainExecutionContext::Operation");
+    }
+
+    match &**execution_error {
+        ExecutionError::SystemError(SystemExecutionError::InsufficientFunding { .. }) => {}
+        _ => panic!("Expected ExecutionError::SystemError::InsufficientFunding"),
+    }
```

Comment:
Idea: since this last `match` doesn't extract anything, it could be an `assert_matches!`.

```suggestion
    assert_matches!(
        execution_error,
        ExecutionError::SystemError(SystemExecutionError::InsufficientFunding { .. }),
        "Expected an `InsufficientFunding` system error, found: {execution_error:#?}"
    );
```